### DISCUSSION TITLE
How can I partially render the TUI?

### DISCUSSION THREAD

**Discussion by Martinits** [ratatui/ratatui#409] [Discussion]
How can I update just part of the widgets for performance?

**Comment by joshka** [comment]
Generally, you can't. The whole approach of Ratatui is to render the entire frame every time draw is called. Because TUIs are fairly small (maximum of 65535 cells), it's doubtful that rendering is likely to be too slow on any reasonably modern hardware.

E.g. a 10K line paragraph [renders < 1ms](https://github.com/ratatui-org/ratatui/pull/262)).
Also note that the algorithm only updates cells that are changed on the actual terminal.

Is there a performance issue that you're seeing? Can you measure it? We'd like to have more benchmarks and understand if there are problematic areas.
